runtime-modules/bounty/src/benchmarking.rs

```diff
@@ -367,17 +503,302 @@ benchmarks! {
             bounty_id
         ).unwrap();
 
-        let bounty = Bounty::<T>::bounties(bounty_id);
-        assert!(matches!(
-            Bounty::<T>::get_bounty_stage(&bounty),
-            BountyStage::Withdrawal {cherry_needs_withdrawal: true, ..}
-        ));
-
-    }: withdraw_creator_funding(RawOrigin::Signed(account_id), creator.clone(), bounty_id)
+    }: withdraw_creator_cherry(RawOrigin::Signed(account_id), creator.clone(), bounty_id)
     verify {
         assert!(!Bounties::<T>::contains_key(bounty_id));
         assert_last_event::<T>(Event::<T>::BountyRemoved(bounty_id).into());
     }
+
+    announce_work_entry {
+        let cherry: BalanceOf<T> = 100.into();
+        let funding_amount: BalanceOf<T> = 100.into();
+        let stake: BalanceOf<T> = 100.into();
+
+        let member_ids = (0.. T::MaxWorkEntryLimit::get())
```

Comment:
I implemented just the worst case. We cannot introduce "let i 0..MaxEntryLimit" here because the extrinsic parameters don't contain any dependency on the underlying state (WorkEntries storage size). We could get this using the special weight function, but it contradicts Parity's recommendation  ("never access the state in the weight function").

Reply:
I think it's still better if we use the 'i in 0..MaxEntryLimit' and then when calculating the weight we pass the `T::MaxWorkEntryLimit::get()` as a parameter like we do [here](https://github.com/Joystream/joystream/blob/olympia/runtime-modules/referendum/src/lib.rs#L452-L454).

This leaves the door open to weight refund if it's ever needed.

Reply:
Ok. Fixed in this [PR](https://github.com/Joystream/joystream/pull/2231)